app/models/user.rb

```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  before_save{self.email = email.downcase}
```

Comment:
```suggestion
  before_save{email.downcase!}
```
phân biệt cho a email.downcase và email.downcase!

Reply:
downcase! là method sửa đổi chuỗi tại chỗ, downcase tạo mới 1 string instance. downcase! trả về nil khi xâu kí tự không có sự thay đổi, không dùng phép downcase! với các câu lệnh khác